File: android/src/test/java/me/carda/awesome_notifications/core/enumerators/MediaSourceTest.java
package me.carda.awesome_notifications.core.enumerators;

import org.junit.Test;

import static org.junit.Assert.*;

public class MediaSourceTest {

    @Test
    public void testEnumOrder() {
        String errorMessage = "";

        assertEquals(errorMessage, 0, MediaSource.Resource.ordinal());
        assertEquals(errorMessage, 1, MediaSource.Asset.ordinal());
        assertEquals(errorMessage, 2, MediaSource.File.ordinal());
        assertEquals(errorMessage, 3, MediaSource.Network.ordinal());
        assertEquals(errorMessage, 4, MediaSource.Unknown.ordinal());
    }
}

File: test/src/enumerators/media_source_test.dart
import 'package:awesome_notifications/src/enumerators/media_source.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  group('MediaSourcePrefix tests', () {
    test('MediaSourcePrefix values are set correctly', () {
      expect(MediaSourcePrefix.Resource, 'resource://');
      expect(MediaSourcePrefix.Asset, 'asset://');
      expect(MediaSourcePrefix.File, 'file://');
      expect(MediaSourcePrefix.Network, 'https://');
      expect(MediaSourcePrefix.Unknown, '');

      expect(MediaSourcePrefix.values,
          containsAll(['resource://', 'asset://', 'file://', 'https://', '']));
    });

    test('MediaSourcePrefix index is set correctly', () {
      expect(const MediaSourcePrefix(1).value, 1);
    });
  });
}
